uniqe.go

```diff
@@ -0,0 +1,32 @@
+// 課題[3]
+// type MyIntSlice []intでIntスライスと互換のある独自型を作り、重複を排除するメソッドを実装する。
+
+// type MyIntSlice []int
+// m := MyIntSlice{1, 2, 2, 3, 3, 3, 4, 5}
+// fmt.Println(m.Unique()) // [1, 2, 3, 4, 5]
+
+package main
+
+import "fmt"
+
+
+type MyIntSlice []int
+
+func main()  {
+	m := MyIntSlice{1, 2, 2, 3, 3, 3, 4, 5}
+
+	fmt.Println(m.Unique())
+}
+
+func (m MyIntSlice) Unique() (unique []int) {
+	validMap := make(map[int]bool)
+	unique = []int{}
+
+	for _, v := range m{
+		if !validMap[v] {
```

Comment:
```suggestion
		if _, ok := validMap[v]; !ok {
```
rubyと違ってgoのmapはこうやって確認できます。これでkeyの有無だけで確認できます。valueはboolでもなんでもOKです。

Reply:
なるほどです。修正しました。